src/provider/provider_tracking.h

```diff
@@ -26,34 +28,61 @@ extern "C" {
 
 extern umf_memory_provider_ops_t UMF_TRACKING_MEMORY_PROVIDER_OPS;
 
+typedef struct tracker_value_t {
+    umf_memory_pool_handle_t pool;
+    size_t size;
+} tracker_value_t;
+
 struct umf_memory_tracker_t {
+    umf_ba_linear_pool_t *pool_linear;
+    umf_ba_pool_t *pool_tracker;
     critnib *map;
-    os_mutex_t *splitMergeMutex;
+    os_mutex_t splitMergeMutex;
 };
 
 typedef struct umf_memory_tracker_t *umf_memory_tracker_handle_t;
 
 static inline umf_memory_tracker_handle_t umfMemoryTrackerCreate(void) {
-    umf_memory_tracker_handle_t handle = (umf_memory_tracker_handle_t)malloc(
-        sizeof(struct umf_memory_tracker_t));
-    if (!handle) {
+    umf_ba_linear_pool_t *pool_linear =
+        umf_ba_linear_create(0 /* minimum pool size */);
+    if (!pool_linear) {
         return NULL;
     }
 
-    handle->map = critnib_new();
-    if (!handle->map) {
-        free(handle);
-        return NULL;
+    umf_ba_pool_t *pool_tracker = umf_ba_create(sizeof(struct tracker_value_t));
```

Comment:
nit: `pool_tracker` sounds like it's something that tracks pools ;d Can we rename it to `tracker_allocator`?

Reply:
Done